core/chains/evm/txmgr/txmgr.go

```diff
@@ -54,13 +54,13 @@ type Config interface {
 }
 
 // KeyStore encompasses the subset of keystore used by txmgr
-type KeyStore interface {
-	CheckEnabled(address common.Address, chainID *big.Int) error
-	EnabledKeysForChain(chainID *big.Int) (keys []ethkey.KeyV2, err error)
-	GetNextNonce(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
-	GetStatesForChain(chainID *big.Int) ([]ethkey.State, error)
-	IncrementNextNonce(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
-	SignTx(fromAddress common.Address, tx *gethTypes.Transaction, chainID *big.Int) (*gethTypes.Transaction, error)
+type KeyStore[ADDR any, ID any, TX any, META any] interface {
+	CheckEnabled(address ADDR, chainID ID) error
```

Comment:
@jmank88 : I know that we need a universal chainId for plugin architecture work.
Is it right to assume that whenever we finalize on that universal-id, we will re-use it here too, for sake of consistency?

Reply:
By here, I mean in the TxMgr component overall